native/src/modules/endpoint/loadFromEndpoint.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import determineApiUrl from './determineApiUrl'
+import { Payload } from 'api-client'
+
+const loadFromEndpoint = async <T>(
```

Comment:
You could refactor this to a hook:

```suggestion
const useEndpointFetch = () => {

   const [data, setData] = useState()
   const [error, setError] = useState()
   
   useEffect(() => {
      //fetch and set state
   })

  return error, data
}
```

Reply:
Thought about that as well, but not sure how I would be able to call it on refresh then? Any ideas?

Reply:
You could create a second hook. Something like:

```suggestion
const useRefresh = (): [boolean, () => void] => {
  const dispatch = useDispatch()
  const range = useSelector<RootState, Range | null>(
    (state) => state.days.range,
  )
  const loading = useSelector<RootState, boolean>((state) => state.days.loading)

  return [
    loading,
    useCallback(() => {
      if (range) {
        dispatch(
          fetchRange({
            from: range.from,
            to: range.to,
          }),
        )
      }
    }, [dispatch, range]),
  ]
}

export default useRefresh

```

Reply:
Okay, sorry, I meant on pull to refresh. E.g. here: https://github.com/Integreat/integreat-app/pull/146/files#diff-d46e02df50fb1c491b623b34aeacad7ca09df8050dc6e8d0634332db3fbf018eR78

Reply:
Or can I call the hook normally even though it contains `useEffect` and `useState`? That sounds weird because we need to share state between both calls...

Reply:
I think in that case it is fine as you did it.